src/mono/wasm/debugger/BrowserDebugProxy/DebugStore.cs

```diff
@@ -389,11 +389,14 @@ public MethodInfo(AssemblyInfo assembly, MethodDefinitionHandle methodDefHandle,
             this.IsEnCMethod = false;
             this.TypeInfo = type;
             DebuggerAttrInfo = new DebuggerAttributesInfo();
-            foreach (CustomAttributeHandle cattr in methodDef.GetCustomAttributes())
+            //we need to loop in all the CustomAttributes from asmMetadataReader because methodDef.GetCustomAttributes() does not work correctly on EnC metadata
+            foreach (CustomAttributeHandle cattr in asmMetadataReader.CustomAttributes)
             {
+                var ca = asmMetadataReader.GetCustomAttribute(cattr);
+                if (ca.Parent.Kind != HandleKind.MethodDefinition || (ca.Parent.Kind == HandleKind.MethodDefinition && ca.Parent.GetHashCode() != (token | (int)TokenType.MdtMethodDef)))
```

Comment:
Could you please add a comment here explaining what this is trying to do?

Reply:
`if (fromEnC && (ca.Parent.Kind != HandleKind.MethodDefinition || ca.Parent.GetHashCode() != (token | (int)TokenType.MdtMethodDef)))`